src/cobra/manipulation/delete.py

```diff
@@ -57,113 +60,40 @@ def prune_unused_reactions(model: "Model") -> Tuple["Model", List["Reaction"]]:
     return output_model, reactions_to_prune
 
 
-def undelete_model_genes(model: "Model") -> None:
-    """Undo the effects of a call to `delete_model_genes` in place.
-
-    Parameters
-    ----------
-    model: cobra.Model
-        The model which will be modified in place.
-
-    """
-    if model._trimmed_genes is not None:
-        for x in model._trimmed_genes:
-            x.functional = True
-
-    if model._trimmed_reactions is not None:
-        for (
-            the_reaction,
-            (lower_bound, upper_bound),
-        ) in model._trimmed_reactions.items():
-            the_reaction.lower_bound = lower_bound
-            the_reaction.upper_bound = upper_bound
-
-    model._trimmed_genes = []
-    model._trimmed_reactions = {}
-    model._trimmed = False
-
-
-def get_compiled_gene_reaction_rules(model: "Model") -> Dict["Reaction", Module]:
-    """Generate a dictionary of compiled gene-reaction rules.
-
-    Any gene-reaction rule expression which cannot be compiled or do not
-    evaluate after compiling will be excluded. The result can be used in the
-    `find_gene_knockout_reactions` function to speed up evaluation of these
-    rules.
-
-    Parameters
-    ----------
-    model: cobra.Model
-        The model to get gene-reaction rules for.
-
-    Returns
-    -------
-    dict of cobra.Reaction, ast.Module
-        The dictionary of cobra.Reaction objects as keys and ast.Module
-        objects as keys.
-
-    .. deprecated::
-        Internal function that has outlived its purpose.
-
-    """
-    warn(
-        "The function `get_compiled_gene_reaction_rules` has outlived its purpose. "
-        "It will be removed soon.",
-        DeprecationWarning,
-    )
-    return {r: r.gpr for r in model.reactions}
-
-
-def find_gene_knockout_reactions(
+def knock_out_model_genes(
```

Comment:
This looks like a major API change and not what the PR title implies. Can you add some more comment to the PR description why that was necessary?

Reply:
Sure.
My reasoning was - delete_model_genes() is marked as deprecated, and should be removed. However, it might be possible people want to set all reactions of a gene to zero. 
I couldn't see any other function doing it, so I added a new function that will do so in a context specific manner. 
If there is one that does it since remove_model_genes removes genes from model, I missed it and would be happy to point to it.

Reply:
All reactions of a gene are already knocked out with a normal `Gene.knockout`. Did you mean knockout several genes in a model? The question there would if we really need a helper since `[g.knockout() for g in gene_list()]` is already pretty concise and will work with the context manager.

Reply:
You're right. Several genes in a model.
My function is simply a helper which returns a list, and so is functionally equivalent to delete_model_genes() only in a context aware manner and with updated code. 
Do we want to have this helper for people who aren't that good with writing python code? If not , I can just delete it.

Reply:
@cdiener - any decision on this?
Keep or remove?